### PR TITLE
Add API token integration

### PR DESCRIPTION
**Pull request by sunkickr**
## Description

Add integration for Astro CLI to work with new API tokens

## 🎟 Issue(s)

- https://github.com/astronomer/astro-cli/issues/1048
- https://github.com/astronomer/astro-cli/issues/1098
- https://github.com/astronomer/astro-cli/issues/1097

## 🧪 Functional Testing

manual testing

## 📸 Screenshots

Screenshot 2023-02-09 at 10 10 14 AM

## 📋 Checklist

- [ ] Rebased from the main (or release if patching) branch (before testing)
- [ ] Ran `make test` before taking out of draft
- [ ] Ran `make lint` before taking out of draft
- [ ] Added/updated applicable tests
- [ ] Tested against [Astro-API](https://github.com/astronomer/astro/) (if necessary).
- [ ] Tested against [Houston-API](https://github.com/astronomer/houston-api/) and [Astronomer](https://github.com/astronomer/astronomer/) (if necessary).
- [ ] Communicated to/tagged owners of respective clients potentially impacted by these changes.
- [ ] Updated any related [documentation](https://github.com/astronomer/docs/)

### CODE REVIEW

File: astro-client/client.go [hunk 1]
```diff
@@ -13,7 +13,7 @@ import (
 )
 
 const (
-	AstronomerConnectionErrMsg = "cannot connect to Astronomer. Try to log in with astro login or check your internet connection and user permissions.\n\nDetails"
+	AstronomerConnectionErrMsg = "cannot connect to Astronomer. Try to log in with astro login or check your internet connection and user permissions. If you are using a API Key or Token make sure your context correct.\n\nDetails"
```
Comment (kushalmalani) [inline astro-client/client.go:16]:
```suggestion
	AstronomerConnectionErrMsg = "cannot connect to Astronomer. Try to log in with astro login or check your internet connection and user permissions. If you are using an API Key or Token make sure your context is correct.\n\nDetails"
```

File: cmd/cloud/setup.go [hunk 2]
```diff
@@ -44,7 +50,20 @@ type TokenResponse struct {
...
+	Version               string   `json:"version"`
+	IsAstronomerGenerated bool     `json:"isAstronomerGenerated"`
+	RsaKeyID              string   `json:"kid"`
+	APITokenID            string   `json:"apiTokenId"`
+	jwt.RegisteredClaims
+}
+
+//nolint:gocognit
 func Setup(cmd *cobra.Command, args []string, client astro.Client, coreClient astrocore.CoreClient) error {
+	// If the user is trying to login or logout no need to go through auth setup.
```
Comment (kushalmalani) [inline cmd/cloud/setup.go:66]: same comment as line 67

File: cmd/cloud/setup.go [hunk 3]
```diff
@@ -79,6 +98,22 @@ func Setup(cmd *cobra.Command, args []string, client astro.Client, coreClient as
...
+	deploymentCmds := []string{"inspect", "create", "update"}
+	if util.Contains(deploymentCmds, cmd.CalledAs()) && cmd.Parent().Use == deploymentCmd {
+		isDeploymentCmd = true
+	}
+
+	// Check for APITokens before API keys or refresh tokens
+	apiToken, err := checkAPIToken(isDeploymentCmd, args)
+	if err != nil {
+		fmt.Println(err)
+		fmt.Println("\nThere was an error using API tokens, using regular auth instead")
```
Comment (kushalmalani) [inline cmd/cloud/setup.go:111]: Considering this is going to be used mostly in ci/cd pipelines where regular auth doesnt exist, should we terminate the flow here?

File: cmd/cloud/setup.go [hunk 4]
```diff
@@ -308,3 +343,71 @@ func checkAPIKeys(astroClient astro.Client, coreClient astrocore.CoreClient, arg
...
 	return true, nil
 }
+
+func checkAPIToken(isDeploymentCmd bool, args []string) (bool, error) {
+	// check os variables
+	astroAPIToken := os.Getenv("ASTRO_API_TOKEN")
+	if astroAPIToken == "" {
+		return false, nil
+	}
+	if !isDeploymentCmd {
```
Comment (kushalmalani) [inline cmd/cloud/setup.go:353]: nit: we dont display "using an astro api token" only on deployment inspect right? maybe we should rename the variable to check `isDeploymentInspect`

File: cmd/cloud/setup.go [hunk 5]
```diff
@@ -308,3 +343,71 @@ func checkAPIKeys(astroClient astro.Client, coreClient astrocore.CoreClient, arg
...
+			return false, err
+		}
+	}
+
+	err = c.SetContextKey("token", "Bearer "+astroAPIToken)
+	if err != nil {
+		return false, err
+	}
+
+	err = c.SetExpiresIn(time.Now().AddDate(1, 0, 0).Unix())
```
Comment (kushalmalani) [inline cmd/cloud/setup.go:386]: Not sure if this is setting the right expiry time. Seeing this locally

` expiresin: 2077-05-26T06:38:54.144956-07:00`

File: cmd/cloud/setup.go [hunk 6]
```diff
@@ -18,20 +18,26 @@ import (
...
 var (
-	authLogin = auth.Login
-
-	client = httputil.NewHTTPClient()
+	authLogin       = auth.Login
+	defaultDomain   = "astronomer.io"
+	client          = httputil.NewHTTPClient()
+	isDeploymentCmd = false
+	parseAPIToken   = util.ParseAPIToken
+	errNotToken     = errors.New("the API token given does not appear to be an Astro API Token")
```
Comment (kushalmalani) [inline cmd/cloud/setup.go:34]: rename `errNotToken` to `errNotApiToken`

File: cmd/cloud/setup.go (hunk 3 above)
Comment (sunkickr) [reply cmd/cloud/setup.go:111]: yes I'll do that

**Pull request merged**